src/hidimstat/desparsified_lasso.py

```diff
@@ -235,99 +179,109 @@ def desparsified_lasso(
+    seed=0,
     n_jobs=1,
-    memory=None,
     verbose=0,
 ):
-    """Desparsified Group Lasso
+    """
+    Desparsified Group Lasso
+
+    Algorithm based Algorithm 1 of d-MTLasso in :cite:`chevalier2020statistical`
+
 
     Parameters
     ----------
     X : ndarray, shape (n_samples, n_features)
-        Data.
+        Input data matrix.
 
     Y : ndarray, shape (n_samples, n_times)
-        Target.
+        Target matrix.
 
     cov : ndarray, shape (n_times, n_times), optional (default=None)
         If None, a temporal covariance matrix of the noise is estimated.
         Otherwise, `cov` is the temporal covariance matrix of the noise.
 
-    test : str, optional (default='chi2')
-        Statistical test used to compute p-values. 'chi2' corresponds
-        to a chi-squared test and 'F' corresponds to an F-test.
-
     max_iter : int, optional (default=5000)
-        The maximum number of iterations when regressing, by Lasso,
-        each column of the design matrix against the others.
+        Maximum number of iterations for nodewise Lasso regressions.
 
     tol : float, optional (default=1e-3)
-        The tolerance for the optimization of the Lasso problems: if the
-        updates are smaller than `tol`, the optimization code checks the
-        dual gap for optimality and continues until it is smaller than `tol`.
-
-    residual_method : str, optional (default='lasso')
-        Method used for computing the residuals of the Nodewise Lasso.
-        Currently the only method available is 'lasso'.
+        Convergence tolerance for optimization.
 
     alpha_max_fraction : float, optional (default=0.01)
-        Only used if method='lasso'.
-        Then alpha = alpha_max_fraction * alpha_max.
+        Fraction of max lambda used for Lasso regularization.
 
-    noise_method : str, optional (default='simple')
-        If 'simple', the correlation matrix is estimated by taking the
-        median of the correlation between two consecutive time steps
-        and the noise standard deviation for each time step is estimated
-        by taking the median of the standard deviations for every time step.
-        If 'AR', the order of the AR model is given by `order` and
-        Yule-Walker method is used to estimate the covariance matrix.
+    noise_method : {'AR', 'simple'}, optional (default='AR')
+        Method to estimate noise covariance:
+        - 'simple': Uses median correlation between consecutive timepoints
+        - 'AR': Fits autoregressive model of specified order
 
     order : int, optional (default=1)
-        If `method=AR`, `order` gives the order of the estimated autoregressive
-        model. `order` must be smaller than the number of time steps.
+        Order of AR model when noise_method='AR'. Must be < n_times.
 
-    n_jobs : int or None, optional (default=1)
-        Number of CPUs to use during the Nodewise Lasso.
+    fit_Y : bool, optional (default=True)
+        Whether to fit Y in noise estimation.
```

Comment:
This is not so clear to me. Could you provide some more details?

Reply:
This is linked to my question 4:
> The function group_reid as an option for "null model", do we keep it or not? (L212 of noise_std.py)

In the Reid function, there is an option to not refit y. I don't understand either what is the usage of this parameter.

Reply:
@bthirion

Reply:
My understanding is than when  `fit_Y` is false, then `y` is taken to be noise only (no regression of X is performed). I don't know why it is there. You can open a specific issue + PR to remove this possibility if it is not used anywhere.

Reply:
see the issue #162 and the PR #163.

Reply:
the option has been removed.